tests/test_cacheable_pdf.py

```diff
@@ -0,0 +1,132 @@
+import zfit
+import zfit.z.numpy as znp
+import tensorflow as tf
+
+from zfit import supports
+from zfit.models.cache import CacheablePDF
+from zfit.util import ztyping
+
+
+class TestPDF(zfit.pdf.ZPDF):
+    _PARAMS = ["alpha"]
+
+    def __init__(self, obs: ztyping.ObsTypeInput, **params):
+        super().__init__(obs, **params)
+        self.pdf_call_counter = tf.Variable(0.0)
+        self.integrate_call_counter = tf.Variable(0.0)
```

Comment:
Yes, that's a good way to go! With this we can count the calls.

However, don't use floats! Use ints. By simply using `0` instead of `0.0` (and/or specify the dtype to be of type int).
The point is that later on we will compare it to counts and floating point comparisons are usually not reliable